js/src/views/edit-attribute-field.js

```diff
@@ -31,6 +31,11 @@ var editAttributeField = Backbone.View.extend( {
 			delete data.placeholder;
 		}
 
+		// Apply default value if one exists and no value is set yet
+		if ( '' === data.value && data.default_value.length > 0 ) {
```

Comment:
Good question. Probably this needs more thought put into the data structure to hold it.

I was only thinking of the case of selects / radio buttons, where it would be easy to make a value required. Text fields where an empty string is a valid choice, but a default value is still useful, add an extra complication.